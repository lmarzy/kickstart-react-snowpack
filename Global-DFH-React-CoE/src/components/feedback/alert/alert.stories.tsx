import React from 'react';
import { Story, Meta } from '@storybook/react';

import { Alert } from './alert';

export default {
  title: 'Components/Feeback/Alert',
  component: Alert,
} as Meta;

export const Default: Story = ({ type, children }) => {
  return <Alert type={type}>{children}</Alert>;
};

Default.args = {
  type: 'info',
  children: 'Alert Component',
};
