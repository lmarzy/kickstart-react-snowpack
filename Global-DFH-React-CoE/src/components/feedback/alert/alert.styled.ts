import styled from 'styled-components';

import { feedbackColors } from '../../../styles/utils/';
import { FeedbackType } from '../../../shared/types-styles';

interface StyledAlertProps {
  $type: FeedbackType;
}

export const StyledAlert = styled.span<StyledAlertProps>`
  display: flex;
  align-items: center;
  height: var(--height-medium);
  padding-left: var(--spacing-2);
  padding-right: var(--spacing-2);
  background-color: ${({ $type }) =>
    `var(--col-${feedbackColors($type).backgroundColor})`};
  border-width: 1px 1px 1px 4px;
  border-style: solid;
  border-color: ${({ $type }) => `var(--col-${feedbackColors($type).borderColor})`};
  border-radius: var(--border-radius);
`;
