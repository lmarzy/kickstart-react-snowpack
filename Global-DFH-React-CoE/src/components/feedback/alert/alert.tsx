import React, { FC } from 'react';

import { FeedbackType } from '../../../shared/types-styles';

import { StyledAlert } from './alert.styled';

export interface AlertProps {
  tag?: 'p' | 'span';
  type: FeedbackType;
}

export const Alert: FC<AlertProps> = ({ tag = 'p', type, children }) => (
  <StyledAlert as={tag} $type={type}>
    {children}
  </StyledAlert>
);
