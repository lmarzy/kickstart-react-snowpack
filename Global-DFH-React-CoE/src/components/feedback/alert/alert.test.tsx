import React from 'react';
import { cleanup, render, screen } from '@testing-library/react';

import { Alert } from './alert';

describe('Alert Component', () => {
  afterEach(cleanup);

  test('renders an info alert', () => {
    render(<Alert type="info">Info Alert</Alert>);

    const alert = screen.getByText('Info Alert');

    expect(alert).toBeInTheDocument();
    expect(alert).toMatchSnapshot();
  });

  test('renders an success alert', () => {
    render(<Alert type="success">Success Alert</Alert>);

    const alert = screen.getByText('Success Alert');

    expect(alert).toBeInTheDocument();
    expect(alert).toMatchSnapshot();
  });

  test('renders an warning alert', () => {
    render(<Alert type="warning">Warning Alert</Alert>);

    const alert = screen.getByText('Warning Alert');

    expect(alert).toBeInTheDocument();
    expect(alert).toMatchSnapshot();
  });

  test('renders an error alert', () => {
    render(<Alert type="error">Error Alert</Alert>);

    const alert = screen.getByText('Error Alert');

    expect(alert).toBeInTheDocument();
    expect(alert).toMatchSnapshot();
  });
});
