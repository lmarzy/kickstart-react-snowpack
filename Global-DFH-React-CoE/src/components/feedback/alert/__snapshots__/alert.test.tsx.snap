// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Component renders an error alert 1`] = `
<p
  class="sc-bdnxRM gbZqVq"
>
  Error Alert
</p>
`;

exports[`Alert Component renders an info alert 1`] = `
<p
  class="sc-bdnxRM jSatBq"
>
  Info Alert
</p>
`;

exports[`Alert Component renders an success alert 1`] = `
<p
  class="sc-bdnxRM hsrOSS"
>
  Success Alert
</p>
`;

exports[`Alert Component renders an warning alert 1`] = `
<p
  class="sc-bdnxRM hTVUlG"
>
  Warning Alert
</p>
`;
